sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -70,10 +69,10 @@ public RetryPolicy(int maxRetries, Duration delayDuration) {
     private boolean shouldRetry(HttpResponse response, int tryCount) {
         int code = response.statusCode();
         return tryCount < maxRetries
-                && (code == HttpResponseStatus.REQUEST_TIMEOUT.code()
-                || (code >= HttpResponseStatus.INTERNAL_SERVER_ERROR.code()
-                && code != HttpResponseStatus.NOT_IMPLEMENTED.code()
-                && code != HttpResponseStatus.HTTP_VERSION_NOT_SUPPORTED.code()));
+                && (code == 408  // Request timeout
```

Comment:
I'm surprised JDK doesn't have an enum of status codes.

Reply:
nit: could we add a private enum for these constants?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html

Reply:
Updated the code to use the HttpURLConnection values.